docs/pages/choose-an-edition/teleport-enterprise/aws-kms.mdx

```diff
@@ -0,0 +1,187 @@
+Instead of the actual private key, Teleport will only store the ID of the KMS
+key.
+In short, private key material will never leave AWS KMS.
+
+If launching a new Teleport cluster this will all be handled during initial
+startup with no specific interventions required after configuration.
+For existing Teleport clusters that already have private key material on disk,
+a CA rotation must be performed.
+Read on to [migrating an existing cluster](#migrating-an-existing-cluster) to
+learn more.
+
+## Prerequisites
+
+The features documented on this page are available in Teleport `15.0.0` and
+higher.
+
+(!docs/pages/includes/commercial-prereqs-tabs.mdx!)
+
+- (!docs/pages/includes/tctl.mdx!)
+
+- An AWS account.
+
+<VarList/>
+
+## Step 1/5. Configure AWS IAM permissions
+
+Your Teleport Auth Service will need permissions to create, sign with, list,
+and destroy KMS keys in your AWS account.
+
+### Create an AWS IAM policy
+
+Start by creating the following AWS IAM policy in your account.
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "ListKeys",
+            "Effect": "Allow",
+            "Action": [
+                "kms:ListKeys"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "CreateKeys",
+            "Effect": "Allow",
+            "Action": [
+                "kms:CreateKey",
+                "kms:TagResource"
+            ],
+            "Resource": "*",
+            "Condition": {
+                "StringEquals": {
+                    "aws:RequestTag/TeleportCluster": "<Var name="teleport.example.com" description="Your Teleport cluster name"/>",
+                    "aws:ResourceTag/TeleportCluster": "<Var name="teleport.example.com" description="Your Teleport cluster name"/>"
+                }
+            }
+        },
+        {
+            "Sid": "UseKeys",
+            "Effect": "Allow",
+            "Action": [
+                "kms:GetPublicKey",
+                "kms:ScheduleKeyDeletion",
+                "kms:DescribeKey",
+                "kms:ListResourceTags",
+                "kms:Sign"
+            ],
+            "Resource": "*",
+            "Condition": {
+                "StringEquals": {
+                    "aws:ResourceTag/TeleportCluster": "<Var name="teleport.example.com" description="Your Teleport cluster name"/>"
+                }
+            }
+        }
+    ]
+}
+```
+
+You can create the policy from the AWS console, or via the AWS CLI.
+
+```code
+$ aws iam create-policy --policy-name TeleportKMS --policy-document file://policy.json
+```
+
+### Attach the AWS IAM policy
+
+Your Teleport Auth Server will need access to an IAM identity with this policy attached.
+You can provide credentials via IAM Roles for Tasks, IAM Roles for EC2,
+credential files, or environment variables.
+See [Specifying Credentials](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#specifying-credentials)
```

Comment:
Would the `docs/pages/includes/aws-credentials.mdx` partial provide this information?

Reply:
yeah, didn't know about that, thanks